Modules/Server/Sources/PocketCastsServer/Public/Cache/EpisodeInfoHandler.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+
+/// Request information about an episode using the show notes endpoint
+public class EpisodeInfoHandler {
+    public static var shared = EpisodeInfoHandler()
+
+    private let showNotesUrlCache: URLCache
+
+    private var requestingNotes = false
+
+    private var showNotesCompletionBlocks: [(ShowNotesPodcast?) -> Void] = []
+    private var showNotesCachedCompletionBlocks: [(ShowNotesPodcast?) -> Void] = []
+
+    private init() {
+        showNotesUrlCache = URLCache(memoryCapacity: 1.megabytes, diskCapacity: 10.megabytes, diskPath: "show_notes")
+    }
+
+    struct ShowNotes: Decodable {
+        let podcast: ShowNotesPodcast
+    }
+
+    struct ShowNotesPodcast: Decodable {
+        let episodes: [ShowNotesEpisode]
+    }
+
+    struct ShowNotesEpisode: Decodable {
+        let uuid: String
+        let showNotes: String
+        let image: String?
+    }
+
+    public func loadShowNotes(podcastUuid: String, episodeUuid: String, cached: ((String) -> Void)? = nil, completion: ((String?) -> Void)?) {
+        var cachedNotes = ""
+        var didSendCachedNotes = false
+
+        requestShowNotes(for: podcastUuid, cached: { showNotes in
+            if let notes = showNotes?.episodes.first(where: { $0.uuid == episodeUuid })?.showNotes {
```

Comment:
Wdyt about reducing the duplicate code by adding a method on the podcast object?

```
    struct ShowNotesPodcast: Decodable {
        let episodes: [ShowNotesEpisode]

        func episode(with uuid: String) -> ShowNotesEpisode? {
            episodes.first(where: { $0.uuid == uuid })
        }
    }
```

Reply:
Done in 78ad1ba29859b9493a9c4829ddf5ab64e4a1b0b1